x-pack/legacy/plugins/code/server/plugin.ts

```diff
@@ -217,7 +214,7 @@ export class CodePlugin {
 
     const { gitOps, lspService } = initLocalService(
       server,
-      this.log,
+      this.initContext.logger,
```

Comment:
I see the problem that we have to pass LoggerFactory down the tree. Hopefully, we can improve this in #39695